pkg/controller/datavolume/pvc-clone-controller.go

```diff
@@ -187,6 +203,19 @@ func (r *PvcCloneReconciler) prepare(syncState *dvSyncState) error {
 	return nil
 }
 
+func (r *PvcCloneReconciler) cleanup(syncState *dvSyncState) error {
+	dv := syncState.dvMutated
+	if err := r.populateSourceIfSourceRef(dv); err != nil {
+		return err
+	}
+
+	if dv.DeletionTimestamp == nil {
+		return nil
+	}
+
+	return r.reconcileVolumeCloneSourceCR(syncState, "PersistentVolumeClaim")
```

Comment:
calling to this function even when not using populators will work but not sure if we should..